### ISSUE TITLE
Potential Issue with Windows OS Paths on opening COMMIT_EDITMSG buffers - Mini.Git

### ISSUE THREAD

**Issue by jayhowey** [echasnovski/mini.nvim#1006]
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)
- [X] I have updated 'mini.nvim' to latest version

### Module(s)

mini.git

### Description

On Windows 10, using Neovim 0.10 stable, the Mini.Git `:Git commit` command gives the following error which might be related to escaping backslashes ( \ ) in Windows paths:

![ErrorMessage](https://github.com/echasnovski/mini.nvim/assets/164809522/cbe9c46a-1a42-47f3-a812-094522acf174)
 
My troubleshooting so far has shown that when no spaces are present in the path to the nvim executable, escaping the backslashes is enough to resolve the problem, allowing the _COMMIT_EDITMSG_ buffer to open:

Adding `editor = string.gsub(editor, "\\","\\\\")` at line 686 _git.lua_ for example can work,

https://github.com/echasnovski/mini.nvim/blob/b8c1d3b2f4cd637760babb8a33ad12fd5afe2ee2/lua/mini/git.lua#L683-L686

However, this fix is incomplete where the path to the nvim executable is something like "C:\Program Files\neovim\bin\nvim.exe" etc. (where the path to nvim.exe contains a space character)

**THANK YOU 1000x+ for MINI.NVIM** - mini is excellent and has replaced all but 2 of the plugins I use (colorschemes). I'm happy to help here however I can. Thanks! 

### Neovim version

0.10

### Steps to reproduce

1. `nvim file_to_edit`
2. `:Git add %`
3. `:Git commit`
4. `:messages` to see the error message again.

### Expected behavior

Mini.Diff can reach the nvim executable, Mini.Diff then opens a new _COMMIT_EDITMSG_ and the (Windows) user edits the commit message as-usual. On a `:wq` the commit is made to the local repository:

![CommitMessageTab](https://github.com/echasnovski/mini.nvim/assets/164809522/80512675-9c17-454e-bd87-4018e136aba8)

### Actual behavior

Neovim :messages area shows an error message, _COMMIT_EDITMSG_ buffer fails to open:

![ErrorMessage](https://github.com/echasnovski/mini.nvim/assets/164809522/cc9bae95-a0e1-41ec-9052-943f1291e678)

**Comment by echasnovski** [comment]
Thanks for the issue!

Yeah, escaping special characters seems to be needed here. As I can not reproduce this on Linux, would you mind testing a solution (similar to yours) in the [`git-escape`](https://github.com/echasnovski/mini.nvim/tree/git-escape) branch? If you could check it with both use case in this issue and the one with spaces (like "C:\Program Files\neovim\bin\nvim.exe" you describe), I'd be really grateful.

**Comment by jayhowey** [comment]
aa06718 Fixes this for me! This commit solves my issue. 

Thank You So Much - I really appreciate your fast attention to this issue and the effort on preparing a fix. This has been a great and encouraging exchange with the benevolent maintainer of mini.nvim.

I confirm that aa06718 fixes #1006, based on the following quick check:

- I obtained a fresh/new installation of  [nvim-win64.msi](https://github.com/neovim/neovim/releases/latest/download/nvim-win64.msi) on a Windows 10 machine. By default, this installs nvim at a file system location containing a space character in its path:_"C:\Program Files\nvim-win64\bin\nvim.exe"_.
- I am using the same environment otherwise, my _init.lua_ and all other involved software (Git for Windows) is held constant.
- I followed the steps from our issue to reproduce my error and.....
- Success! The COMMIT_EDITMSG buffer behaves as expected.
